crates/task-impls/src/vid.rs

```diff
@@ -179,88 +80,10 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
-                let collection_view =
-                    if let Some((collection_view, collection_id, _)) = &self.vote_collector {
-                        // TODO: Is this correct for consecutive leaders?
-                        if view > *collection_view {
-                            // warn!("shutting down for view {:?}", collection_view);
-                            self.registry.shutdown_task(*collection_id).await;
-                        }
-                        *collection_view
-                    } else {
-                        TYPES::Time::new(0)
-                    };
-
-                if view > collection_view {
-                    let new_accumulator = VoteAccumulator {
-                        vote_outcomes: HashMap::new(),
-                        sig_lists: Vec::new(),
-                        signers: bitvec![0; self.membership.total_nodes()],
-                        phantom: PhantomData,
-                    };
-
-                    let accumulator = new_accumulator.accumulate(&vote, self.membership.as_ref());
-
-                    let state = VIDVoteCollectionTaskState {
-                        network: self.network.clone(),
-                        membership: self.membership.clone(),
-                        public_key: self.public_key.clone(),
-                        private_key: self.private_key.clone(),
-                        accumulator,
-                        cur_view: view,
-                        event_stream: self.event_stream.clone(),
-                        id: self.id,
-                    };
-                    let name = "VID Vote Collection";
-                    let filter = FilterEvent(Arc::new(|event| {
-                        matches!(event, HotShotEvent::VidVoteRecv(_))
-                    }));
-                    let builder =
-                        TaskBuilder::<VIDVoteCollectionTypes<TYPES, I>>::new(name.to_string())
-                            .register_event_stream(self.event_stream.clone(), filter)
-                            .await
-                            .register_registry(&mut self.registry.clone())
-                            .await
-                            .register_state(state)
-                            .register_event_handler(handle_event);
-                    let id = builder.get_task_id().unwrap();
-                    let stream_id = builder.get_stream_id().unwrap();
-                    let _task = async_spawn(async move {
-                        VIDVoteCollectionTypes::build(builder).launch().await
-                    });
-                    self.vote_collector = Some((view, id, stream_id));
-                } else if let Some((_, _, stream_id)) = self.vote_collector {
-                    self.event_stream
-                        .direct_message(stream_id, HotShotEvent::VidVoteRecv(vote))
-                        .await;
-                };
-            }
             HotShotEvent::VidDisperseRecv(disperse, sender) => {
```

Comment:
Now that we have this event handler in the consensus task, we can remove it entirely from the vid task.